[DOC IMP]Add hardware boot camp contents.

silabs-junfan opened the pull request:
Description about the pull request:
Add hardware boot camp contents.

Review:
@MarkDing 
@ChengYuan-CY

ChengYuan-CY commented:
@silabs-junfan  Thanks a lot for the contribution, below are few comments on my side.
1.
The title of the pull request should have a prefix, for e.g., [DOC IMP].  
https://github.com/MarkDing/IoT-Developer-Boot-Camp/blob/master/CONTRIBUTING.md#pull-requests

2.
Please new a folder "images" under the "Hardware", and then the image link in the markdown will be similar as "images/HW-Schematic-Design/xxx.png"

3. Comments on the "Wireless Products, Protocols and Applications"
It's a very good article to summarize Silicon Labs wireless protocols, products and the typical applications.
a)
Please add TOC (table of content) in the beginning of this article.
b)
Indent the "NameConvention-1.png" and section below to keep align with the previous one.
"As you can see from the name convention, the proprietary chips include..."
c)
There are lots of legacy products for proprietary and Zigbee, a very brief introduction to those products is helpful for user.
d)
It's better to describe the Bluetooth Xpress produce a little bit more since we mentioned it in the section "BLE products and applications".
e)
Add some introduction about the BGM or MGM modules to offer benefit to customer for selecting the right products.
f) Home Automation, Beaconing, Sport and Fitness, Asset Tracking, Classic Audio Stream, AoA/AoD, etc.
As far as I know, the Blue Gecko (EFR32 and the modules base on them) are not intended for audio. At this moment, Bluetooth SIG does not have a GATT profile for audio over BLE.

4. Comments on the "Schematic Design"
a)
Please add TOC (table of content) in the beginning of this article.
b)
How about rename this article as "Schematic Design Guide" ?
c)
Proprietary-Page.png
We can just capture the web page (Alt+print screen) rather than the whole screen.
d) you can refer to the following application note for detail descriptions:AN0002.1
Customer can refer to AN0002.1 and AN0002.2 for Wireless Gecko Series 1 and Wireless Gecko Series 2 products hardware design respectively.
e)
Maybe we can emphasize that the RESETn must not be connected through an external pull-up to an active supply or otherwise driven high as this could damage the device. (4.3 of AN0002.1)
f) For additional information on the external oscillators, refer to the application note, AN0016.2
For additional information on the external oscillators, refer to the application note AN0016.1 or AN0016.2
g) there are 2 application notes described the methods. One is for sub-GHz band, and the other is for 2.4GHz band.
It's better if we can provide the ANs link here.
h) Few minor typo, and I also leave the comment in the specified line of the markdown.
and down load the reference design package to 
Therefor Silabs provides a lot of application notes or reference manue for customer reference.
both the LF and HF domains areexter nal oscillators
Low Frequency Clock Sources. (I think the second "Low Frequency Clock Sources" should be "High Frequency Clock Sources")

ChengYuan-CY commented on Hardware/Schematic Design.md:
```diff
@@ -0,0 +1,143 @@
+![](../images/HW-Schematic-Design/Proprietary-S1.png)
+In this page, you can find the brief introduction of the proprietary series 1 chips and the parameter list of each chips. Ignore the chips list, you can click the View Document in this page and acess the documents list pages directly:
+![](../images/HW-Schematic-Design/Proprietary-Reference.png)
+For example, If you want to look for 434MHz +10dBm output power reference design, you can click item EFR32FG1 BRD4251B 2400/433 MHz 10dBm Radio Board Full Design Package and down load the reference design package to your local address.
```
download

ChengYuan-CY commented on Hardware/Schematic Design.md:
```diff
@@ -0,0 +1,143 @@
+
+But keep in mind that this reference design can be applied to a cluster of chips that have difference packages and frequency band configurations. So you can just replace the chips to your interest chip with different package and frequency band.
+## Hardware design considerations
+After you get the reference design, you may have the concern why the reference is designed like this. Therefor Silabs provides a lot of application notes or reference manue for customer reference. For basical hardware considerations, you can refer to the following application note for detail descriptions:[AN0002.1](https://www.silabs.com/documents/public/application-notes/an0002.1-efr32-efm32-ser ...
```
typo, therefore, reference manual

ChengYuan-CY commented on Hardware/Schematic Design.md:
```diff
@@ -0,0 +1,143 @@
+
+- External clock source:
+
+  EFR32 Wireless Gecko Series devices support different external clock sources to generate the high and low frequency clocks in addition to the internal LF and HF RC oscillators. The possible external clock sources for both the LF and HF domains areexter nal oscillators (square or sine wave) or crystals/ceramic resonators.
```
typo: are external

ChengYuan-CY commented on Hardware/Schematic Design.md:
```diff
@@ -0,0 +1,143 @@
+
+    An external low frequency clock can be supplied from a crystal/ceramic resonator or from an external clock source.It can source a low-frequency clock from an external source such as a TCXO or VCXO.
+  ![](../images/HW-Schematic-Design/LF-Connection.png)
+  - Low Frequency Clock Sources
```
typo: High

ChengYuan-CY commented on Hardware/Schematic Design.md:
```diff
@@ -0,0 +1,143 @@
+
+  - Low Frequency Clock Sources
+
+    An external low frequency clock can be supplied from a crystal/ceramic resonator or from an external clock source.It can source a low-frequency clock from an external source such as a TCXO or VCXO.
```
typo: A space should be added after the period "."

silabs-junfan commented:
> @silabs-junfan Thanks a lot for the contribution, below are few comments on my side.
> 1.
> The title of the pull request should have a prefix, for e.g., [DOC IMP]. You can get more information about the rule from the [CONTRIBUTING.md](https://github.com/MarkDing/IoT-Developer-Boot-Camp/blob/master/CONTRIBUTING.md#pull-requests).
Answer: Yes, modified.
> Please new a folder "images" under the "Hardware", and then the image link in the markdown will be similar as "images/HW-Schematic-Design/xxx.png"
Answer: Modified.
> 1. Comments on the "Wireless Products, Protocols and Applications"
>    It's a very good article to summarize Silicon Labs wireless protocols, products and the typical applications.
>    a)
>    Please add TOC (table of content) in the beginning of this article.
Answer: Added.
>    b)
>    Indent the "NameConvention-1.png" and section below to keep align with the previous one.
>    "As you can see from the name convention, the proprietary chips include..."
Answer: Modified
>    c)
>    There are lots of legacy products for proprietary and Zigbee, a very brief introduction to those products is helpful for user.
Answer: This training aim to EFR32xG1x series chip.
>    d)
>    It's better to describe the Bluetooth Xpress produce a little bit more since we mentioned it in the section "BLE products and applications".
Answer: The training purpose is to teach people how to find Silabs' reference design, and to do customized design based on the reference. Just use EFR32xG1x series chip as example. And will not mention others.
>    e)
>    Add some introduction about the BGM or MGM modules to offer benefit to customer for selecting the right products.
Answer: The training purpose is to teach people how to find Silabs' reference design, and to do customized design based on the reference. Just use EFR32xG1x series chip as example. And will not mention others.
>    f) Home Automation, Beaconing, Sport and Fitness, Asset Tracking, Classic Audio Stream, AoA/AoD, etc.
>    As far as I know, the Blue Gecko (EFR32 and the modules base on them) are not intended for audio. At this moment, Bluetooth SIG does not have a GATT profile for audio over BLE.
Answer: Modified
> 2. Comments on the "Schematic Design"
>    a)
>    Please add TOC (table of content) in the beginning of this article.
Answer: Added. 
>    b)
>    How about rename this article as "Schematic Design Guide" ?
Answer: Modified
>    c)
>    Proprietary-Page.png
>    We can just capture the web page (Alt+print screen) rather than the whole screen.
Answer: Modified
>    d) you can refer to the following application note for detail descriptions:AN0002.1
>    Customer can refer to AN0002.1 and AN0002.2 for Wireless Gecko Series 1 and Wireless Gecko Series 2 products hardware design respectively.
Answer: Modified
>    e)
>    Maybe we can emphasize that the RESETn must not be connected through an external pull-up to an active supply or otherwise driven high as this could damage the device. (4.3 of AN0002.1)
Answer: Modified
>    f) For additional information on the external oscillators, refer to the application note, AN0016.2
>    For additional information on the external oscillators, refer to the application note AN0016.1 or AN0016.2
Answer: Modified
>    g) there are 2 application notes described the methods. One is for sub-GHz band, and the other is for 2.4GHz band.
>    It's better if we can provide the ANs link here.
Answer: Modified
>    h) Few minor typo, and I also leave the comment in the specified line of the markdown.
>    and down load the reference design package to
Answer: Modified
>    Therefor Silabs provides a lot of application notes or reference manue for customer reference.
Answer: Modified
>    both the LF and HF domains areexter nal oscillators
Answer: Modified
>    Low Frequency Clock Sources. (I think the second "Low Frequency Clock Sources" should be "High Frequency Clock Sources")
Answer: Modified
